### PR TITLE
Admin Generator (Future): Add  `FinalFormRangeInput` to form generator

### PR DESCRIPTION
**Pull request by andrearutrecht**
## Description
`FinalFormRangeInput` fields can now be generated in forms. The new field type is `numberRange`. Data of the field is saved in this format: `{ min: number, max: number }` 

The names `min` and `max` are currently hardcoded, as they **have to** be named this way. `FinalFormRangeInput` passes the values named like this.

## Example

The field with all possible options:

```
{ 
     type: "numberRange",
     name: "priceRange", 
     minValue: 25, // required
     maxValue: 500, // required
     disableSlider: true, //  optional, default is false
     startAdornment: "€" // optional
     endAdornment: "€" // optional, both adornments can be added for a field
 }
```

The field in product.entity:

```
@ObjectType()
@InputType("ProductPriceRangeInput")
export class ProductPriceRange {
    @Field()
    @IsNumber()
    min: number;

    @Field()
    @IsNumber()
    max: number;
}

{...}

export class Product extends BaseEntity<Product, "id"> {

{...}

    @Property({ type: "json", nullable: true })
    @Field(() => ProductPriceRange, { nullable: true })
    priceRange?: ProductPriceRange = undefined;

{...}

}
```

-   [x] I have verified if my change requires an example

Example in demo: https://github.com/vivid-planet/comet/pull/2593

## Screenshots/screencasts

Creating a Product with a price range:

https://github.com/user-attachments/assets/bde61e5c-7f9e-4ae6-a4a7-bd54ea7d33d9

Example of editing a price range:

https://github.com/user-attachments/assets/52ea6f9e-115b-4714-8416-a32be5dc6efd

## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents

Jira Task: https://vivid-planet.atlassian.net/browse/SVK-124

### CODE REVIEW

File: demo/admin/src/products/future/generated/ProductForm.tsx [hunk 1]
```diff
@@ -131,6 +136,7 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
         const output = {
             ...formValues,
             category: formValues.category?.id,
+            priceRange: formValues.priceRange ? { min: parseFloat(formValues.priceRange.min), max: parseFloat(formValues.priceRange.max) } : null,
```
Comment (nsams) [inline demo/admin/src/products/future/generated/ProductForm.tsx:139]: why this conversion (also for mutation)?

you also didn't need it in #2593 (which should be the base for this)

Comment (andrearutrecht) [reply demo/admin/src/products/future/generated/ProductForm.tsx:139]: You're right, I don' t need it. I will remove it.

File: packages/admin/cms-admin/src/generator/future/generator.ts [hunk 2]
```diff
@@ -35,6 +36,14 @@ type MultiFileFormFieldConfig = { type: "fileUpload"; multiple: true; maxFiles?:
 export type FormFieldConfig<T> = (
     | { type: "text"; multiline?: boolean }
     | { type: "number" }
+    | {
+          type: "numberRange";
+          minValue: number;
+          maxValue: number;
+          disableSlider?: boolean;
+          startAdornment?: ReactNode;
+          endAdornment?: ReactNode;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/generator/future/generator.ts:45]: Should we add adornment support as a general feature for all field types?

Comment (nsams) [reply packages/admin/cms-admin/src/generator/future/generator.ts:45]: Why not, especially for number input

For normal text input i dont see a use case, but the config option also wont do any harm

File: packages/admin/cms-admin/src/generator/future/generator.ts [hunk 3]
```diff
@@ -35,6 +36,14 @@ type MultiFileFormFieldConfig = { type: "fileUpload"; multiple: true; maxFiles?:
 export type FormFieldConfig<T> = (
     | { type: "text"; multiline?: boolean }
     | { type: "number" }
+    | {
+          type: "numberRange";
+          minValue: number;
+          maxValue: number;
+          disableSlider?: boolean;
+          startAdornment?: ReactNode;
```
Comment (nsams) [inline packages/admin/cms-admin/src/generator/future/generator.ts:44]: You can not use abritary react components in generator config, please allow only string

File: packages/admin/cms-admin/src/generator/future/generator.ts (hunk 2 above)
Comment (andrearutrecht) [reply packages/admin/cms-admin/src/generator/future/generator.ts:45]: Should I add that here or is that something you'd rather want in its own PR?

Comment (johnnyomair) [reply packages/admin/cms-admin/src/generator/future/generator.ts]: Please do this in a follow-up PR

File: packages/admin/cms-admin/src/generator/future/generator.ts (hunk 3 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/src/generator/future/generator.ts]: I can think of other use cases for the adornment, e.g. icons. But we can add this later. String is good for now.

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**